openedx_tagging/core/tagging/rest_api/v1/serializers.py

```diff
@@ -175,3 +176,33 @@ class TaxonomyTagDeleteBodySerializer(serializers.Serializer):  # pylint: disabl
         child=serializers.CharField(), required=True
     )
     with_subtags = serializers.BooleanField(required=False)
+
+
+class TaxonomyImportBodySerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer of the body for the Taxonomy Import action
+    """
+    taxonomy_name = serializers.CharField(required=True)
+    taxonomy_description = serializers.CharField(default="")
+    file = serializers.FileField(required=True)
+
+    def get_parser_format(self, obj):
+        """
+        Returns the ParserFormat based on the file extension
+        """
+        filename = obj["file"].name
+        ext = filename.split('.')[-1]
+        if ext.lower() == 'csv':
+            return ParserFormat.CSV
+        elif ext.lower() == 'json':
+            return ParserFormat.JSON
```

Comment:
nit: this can be made more general:

```suggestion
        parser_format = getattr(ParserFormat, ext.upper(), None)
        if parser_format:
            return parser_format
```